win_etw_macros/src/lib.rs

```diff
@@ -1170,30 +1219,61 @@ fn parse_event_attributes(
                                         lit: lit.clone(),
                                         attrs: Vec::new(),
                                     });
+                                } else if *path == parse_quote!(id) {
+                                    if event_id.is_some() {
+                                        return Err(Error::new(
+                                            lit.span(),
+                                            "Event id has already been defined.".to_string(),
+                                        ));
+                                    }
+                                    if let Lit::Int(lit_int) = lit {
+                                        if let Ok(int_value) = lit_int.base10_parse() {
+                                            // eprintln!("{}", int_value);
+                                            event_id = Some(int_value);
+                                        } else {
+                                            return Err(Error::new(
+                                                lit.span(),
+                                                "Event id must be a u16.".to_string(),
+                                            ));
+                                        }
+                                    } else {
+                                        return Err(Error::new(
+                                            lit.span(),
+                                            "Event id must be a u16.".to_string(),
+                                        ));
+                                    }
                                 } else {
-                                    errors
-                                        .push(Error::new_spanned(item, "Unrecognized attribute."));
+                                    return Err(Error::new(
+                                        lit.span(),
+                                        format!("Unrecognized attribute: {}.", lit_to_string(lit)),
+                                    ));
                                 }
                             }
                             _ => {
-                                errors.push(Error::new_spanned(item, "Unrecognized attribute."));
+                                return Err(Error::new(
+                                    item.span(),
+                                    format!("Unrecognized attribute: {:?}", item),
```

Comment:
Same thing here.  Drop the `{:?}` and the `item` parameter, and let the `Span` do the heavy lifting for us.